### ISSUE TITLE
mmol/l unchecked constantly

### ISSUE THREAD

**Issue by alexey6600** [pachi81/GlucoDataHandler#115]
Option "Use mmol/l" in Settings-> General menu, constantly unchecking. 
For last day- twise. 
Using last version from Play store ( 0.9.8).

PS. Big thanks for you work!

**Comment by pachi81** [comment]
Which source are you using?
For xDrip and nightscout I already have created a bugfix:
https://onedrive.live.com/?authkey=%21AG8eUfvdnCAQ1G4&id=F01898105C58ADFA%211299367&cid=F01898105C58ADFA

Can you check it? If it occurs again, can you make a screenshot, where I can see the value?
Thank you!

**Comment by alexey6600** [comment]
Source is XDrip+.
Thanks.  I'll try big fix

**Comment by pachi81** [comment]
Really strange! Your are the second one with this issue. The other one had the same problem, but mmol was checked once a day and she is using mg/dl.
Maybe there is a special broadcast, which causes the problem.
Have you detected a special value, when it happened?
I think I have to remove the auto-detection. Only for the first value...

**Comment by alexey6600** [comment]
ActuallyI just installed it yesterday. And not detect a special value, when it happened.
Just installed version 0.9.8.3, and keep watching.
Thanks a lot! 
Really good software in combination with Wear watch!

**Comment by pachi81** [comment]
Thank you for testing! Is it hard to fix something, I can not reproduce...
But I have changed the handling dor xDrip, that it should not happen. 
But I'm still not sure, what causes the problem...

**Comment by alexey6600** [comment]
I will definitely write about the test results!

**Comment by alexey6600** [comment]
With new version there are no problems with unchecked option.
But found new issue (on attachment )
![Screenshot_20240109_154248_xDrip+](https://github.com/pachi81/GlucoDataHandler/assets/15729909/f7d00932-048d-4c18-9792-30182ea5bf4f)

**Comment by pachi81** [comment]
What happened? The screenshot is gone!? Maybe that was the reason for the problem with unchecking. There seems to be a message without value. I also have activated xDrip as source, but have not had this problem until now.
I also check in the next version (not online, yet) also for the existence of the value.

**Comment by alexey6600** [comment]
Removed, as problem gone. 
But problem appeared again. 
![Screenshot_20240109_154248_xDrip+](https://github.com/pachi81/GlucoDataHandler/assets/15729909/3c270808-476a-43f9-9fae-0fbb2739def9)

**Comment by pachi81** [comment]
Do you know how to use logcat or do you have any logcat client on your phone?

**Comment by alexey6600** [comment]
Know only with ABD

**Comment by pachi81** [comment]
Can you create logcat for my app? Maybe it contains the problem, just have a look, if the time of the problem is inside...

**Comment by alexey6600** [comment]
Maybe there are application for this ?

**Comment by alexey6600** [comment]
[logcat.txt](https://github.com/pachi81/GlucoDataHandler/files/13874923/logcat.txt)

**Comment by pachi81** [comment]
Strange, it contains a lot of logs of my app, but only from Android System, not from my app itself.
I think you have to give the app the permission to read all logs.

`adb shell "pm grant com.dp.logcatapp android.permission.READ_LOGS && am force-stop com.dp.logcatapp"`

See here for more details: https://github.com/darshanparajuli/LogcatReader

**Comment by alexey6600** [comment]
I used adb with grep on macos) It's simpier)

**Comment by pachi81** [comment]
Try `adb shell logcat | grep GDH.`

**Comment by pachi81** [comment]
Or using pid (not tested!)
```
adb logcat --pid=`adb shell pidof -s de.michelinside.glucodatahandler`
```

**Comment by alexey6600** [comment]
[logcat2.txt](https://github.com/pachi81/GlucoDataHandler/files/13875279/logcat2.txt)

**Comment by pachi81** [comment]
Great! Now, I see the values of my app. Unfortunately only the last half an hour. 
Can you create it again, if the issue occurs again? 
And as you broadcast looks different to my one, which source do you use in xDrip+?

**Comment by alexey6600** [comment]
Ok. When issue occurs I create logcat again.
In XDrip source is Sync Follower

**Comment by alexey6600** [comment]
Happened again
[logcat3.txt](https://github.com/pachi81/GlucoDataHandler/files/13876670/logcat3.txt)

**Comment by pachi81** [comment]
Great! Thank you!
I have received these data:
`01-09 18:46:35.785 20911 20911 I GDH.XDripBroadcastReceiver: Glucose: 0.0 - Time: 18:46:35 - Slope: 0.0 - SlopeName: null - Raw: 172.88371999999998 - Source: null (Follower)`
What has xDrip shown at 18:46:35? The raw value is set to 9,6 mmol/l.
Is that correct.

At the moment I'm not sure, what I will do, either ignore or using the raw value.

The raw value I do not really understand, for example I always get a raw value of 0.0.

I have to check other sources.

**Comment by alexey6600** [comment]
![Screenshot_20240109_201603_xDrip+](https://github.com/pachi81/GlucoDataHandler/assets/15729909/aa887b1e-1602-4ecb-af83-0318f2d8add2)

**Comment by pachi81** [comment]
Thank you very much!
I use the raw value, but I do not have trend information, so I will display a ?
You can download this version 0.9.8.5 here:
https://onedrive.live.com/?authkey=%21AG8eUfvdnCAQ1G4&id=F01898105C58ADFA%211299367&cid=F01898105C58ADFA

**Comment by alexey6600** [comment]
Problem appeared again with new version
![Screenshot_20240110_135029_xDrip+](https://github.com/pachi81/GlucoDataHandler/assets/15729909/a92dd569-5547-4fd3-8295-23dce075afc1)

**Comment by pachi81** [comment]
:-(
Can you provide logcat?

**Comment by alexey6600** [comment]
[logcat4.txt](https://github.com/pachi81/GlucoDataHandler/files/13887198/logcat4.txt)

**Comment by alexey6600** [comment]
I noticed that such problems arise with a sharp increase or decrease in sugar levels

**Comment by pachi81** [comment]
Have you already created an issue for xDrip? If not I can also do it.
Here is the data, I have received:
```
{ 
  com.eveningoutpost.dexdrip.Extras.NoiseBlockLevel => 200
  com.eveningoutpost.dexdrip.Extras.VersionInfo => 2123092521 ab4c6032b-2023.09.25
  com.eveningoutpost.dexdrip.Extras.Raw => 175.88371999999998
  com.eveningoutpost.dexdrip.Extras.Time => 1704887351817
  com.eveningoutpost.dexdrip.Extras.CalibrationInfo => 1704748394164
  com.eveningoutpost.dexdrip.Extras.NsNoiseLevel => NULL
  com.eveningoutpost.dexdrip.Extras.Noise => 250.99240296515094
  com.eveningoutpost.dexdrip.Extras.SensorBattery => 0
  com.eveningoutpost.dexdrip.Extras.SourceDesc => Follower
}
```

There is missing some informations, like `com.eveningoutpost.dexdrip.Extras.BgSlope`, `com.eveningoutpost.dexdrip.Extras.BgSlopeName` and `com.eveningoutpost.dexdrip.Extras.BgEstimate`.

At the moment, I check for the presence of BgSlope. I can also remove this to show at least the raw-value without trend direction...

**Comment by pachi81** [comment]
I have checked the source code of xDrip and it seems to be not a bug, it is a "feature".
The problem is the NoiseBlockLevel which is set to 200 on your device. The current noise level is around 250.
So xDrip blocks these levels and do not send any estimated values (even the trend).

Please go to settings - inter app settings and change the noise block level (3. entry) to send anything.

**Comment by alexey6600** [comment]
Done.
Thanks! I'll test, and write about the test results!

**Comment by pachi81** [comment]
Hopefully, you will not detect it, as all should work like expected, now.
I will create a hint for this problem in GDH, if it occurs.

Thank you so much for your help!!!

**Comment by pachi81** [comment]
Hi!
Does it happened again?

**Comment by alexey6600** [comment]
Hello.
No. Work well!
Thanks!

Ps. A small wish, if possible)
On the watch, make the option "glucose and trend" colored, and not big.
Now there is such an option, but it’s a big one.

**Comment by pachi81** [comment]
Great to hear. 
Can you do me one more favor, can you install the last beta version 0.9.8.6-dev and change the xDrip Noise Block settings to lowest one to force this error happens again? Than you should receive a question mark instead of the trend arrow and there should be some hint on main screen of my app. 
After you have received a valid value, this hint should disappear... 

I tried to reproduce this issue, but for my source the highest noise was around 4. So I did not get into this problem.

> Ps. A small wish, if possible)
On the watch, make the option "glucose and trend" colored, and not big.
Now there is such an option, but it’s a big one.

Wow, I did not expect such a whish... The most people like to have it very big. The normal, small ones are no images and I can only add colors to images. But I can limit the font size while creating the images, so it would be smaller. 
What do you expect, only the glucose value to be smaller or both?

**Comment by alexey6600** [comment]
With big ones I have this:
![Screenshot_20240112_153522_sysui](https://github.com/pachi81/GlucoDataHandler/assets/15729909/b54e69df-0820-406f-9bd7-80e2387a289e)

With normal more nice look:
![Screenshot_20240112_153500_sysui](https://github.com/pachi81/GlucoDataHandler/assets/15729909/2b4bbdcf-fb45-40c3-b87f-1e3031588950)

**Comment by alexey6600** [comment]
Yep. With latest dev there are question mark and hint. 
![Screenshot_20240112_154429_GlucoDataHandler](https://github.com/pachi81/GlucoDataHandler/assets/15729909/6cc0e74a-6ee4-4c08-b7a0-7355a56f4c58)

**Comment by pachi81** [comment]
Yes the normal is text only with icon, where I can not change the color, but I can add an additional image with smaller size.

**Comment by pachi81** [comment]
Great, that is what I expected!

**Comment by pachi81** [comment]
Hi,
I created a new version with small colored complications. Can you check?
You can also join my beta playstore for easier installation on watch, but you know, how to use adb.

**Comment by alexey6600** [comment]
Thanks!
I'm allready joined to beta on playstore. But I don't see a newer version yet

**Comment by pachi81** [comment]
Don't know, how it works. At first step, I added a mail list. But there are 3 areas until release, maybe you will are on a higher layer...
But for the higher layers, I need google confirmation.

**Comment by pachi81** [comment]
I moved it on a higher level. Now I have to wait for google...
If I have it, maybe you will receive it as beta tester...

**Comment by alexey6600** [comment]
Ok. Thanks a lot!
I will wait.

**Comment by alexey6600** [comment]
Just received update. 
Thanks! Nice look now on watch.

**Comment by pachi81** [comment]
Can you make a screenshot? 
On my watch it looks different to the emulator.
So I'd like to see, how it looks on your watch.
Thank you!

**Comment by alexey6600** [comment]
![Screenshot_20240114_095636_sysui](https://github.com/pachi81/GlucoDataHandler/assets/15729909/3d46140d-8c1e-4121-b60d-b7eeb8d2ae37)

**Comment by alexey6600** [comment]
Option "foreground" disappeared. 
This is normal?

**Comment by pachi81** [comment]
It looks like on my watch....
In the emulator I have less space.

Yes, I removed the foreground option as some user deactivated it and complaint about problems...
Now it is mandatory for every one.

**Comment by pachi81** [comment]
@alexey6600 I create a new beta and uploaded it. After google accepted it, you will get the new version.
I changed the small trend-glucose complication to have less space between the value and the trend arrow.
Please provide some feedback and screenshot, how do you like. 
Thank you!

**Comment by alexey6600** [comment]
Thanks!
After I receive update - give you feedback!

**Comment by alexey6600** [comment]
Is version 0.9.8.8 (on watch) new ?
Now is very small space between value and trend. 
![Screenshot_20240118_162456_weather](https://github.com/pachi81/GlucoDataHandler/assets/15729909/59b4faed-6584-4030-a582-f99c263d59fc)

**Comment by pachi81** [comment]
Update is available, now...

**Comment by alexey6600** [comment]
![Screenshot_20240118_183153_sysui](https://github.com/pachi81/GlucoDataHandler/assets/15729909/2b592973-1a48-48b9-9ab9-e36b16356d65)

**Comment by pachi81** [comment]
Is it ok? Or still to big?

**Comment by alexey6600** [comment]
Still too big. But it can be like that)

**Comment by pachi81** [comment]
Ok, because, the next size, which looks good, I think is too small.
So I would let it be like that...

**Comment by alexey6600** [comment]
Thank you! 
Really good software?

**Comment by pachi81** [comment]
Thank you!
On my watch the small is nearly the same like the normal one:
![Screenshot_20240118_185743_glucodatahandler~2](https://github.com/pachi81/GlucoDataHandler/assets/56817032/c02eda7f-ef8c-40b6-abbf-8d3d75366dab)

**Comment by pachi81** [comment]
Fixed with version 0.9.9
